### PR TITLE
Fix some items not being addable to artifact bags of holding

### LINKED ISSUE #16834: Artifact Bag of Holding and multitools
### Labels

[BUG][MINOR]

### Description

Multitools won't go into activated bags of holding, even if they can fit. (doesn't give cannot fit message)

### Steps to reproduce

1. Acquire a bag of holding
2. try putting a multitool in it

### Expected Behavior

Well it should work but doesn't! I expected as much though.

### Additional Information & Screenshots

Reported by: Poglodyte
On server: Goonstation 1 Classic: Heisenbee
Active test merges: #16744 and #16370
Round log date: 2023-11-08-02-56
Reported on: 2023-11-08 03:42:43
Map: COGMAP

### PR DESCRIPTION
**Pull request by FlameArrow57**
[GAME OBJECTS][PLAYER ACTIONS][BUG]

## About the PR 
This PR fixes some items, like multitools, not being able to be added to artifact bags of holding.

Also, activator keys will now have the same effect when added to a bag of holding with an origin they can activate as adding another bag of holding.

## Why's this needed? 
Bug fix, fixes #16834

## Changelog 

```changelog
(u)FlameArrow57
(+)It's no longer advised to put an activator key into a bag of holding it can activate!
```

### CODE REVIEW

File: code/obj/artifacts/artifactprocs.dm [hunk 1]
```diff
@@ -263,7 +263,11 @@
 						for(var/i=1,i<rand(1,3),i++)
 							src.ArtifactDevelopFault(100)
 				else if (A.activated)
-					src.ArtifactDeactivated()
+					if (istype(src, /obj/item/artifact/bag_of_holding))
+						if (!user || user.a_intent == INTENT_HARM)
+							src.ArtifactDeactivated()
```
Comment (Tarmunora) [inline code/obj/artifacts/artifactprocs.dm:268]: Intent-based interactions kinda suck. Would rather just have deactivators not be storable at all in bags of holding that they'd deactivate

**Review by Tarmunora — CHANGES_REQUESTED** [review]

File: code/obj/artifacts/artifactprocs.dm (hunk 1 above)
Comment (FlameArrow57) [reply code/obj/artifacts/artifactprocs.dm:268]: Sure. Alternate idea, maybe allow them to be added by clicking the activator key with the bag of holding?

Comment (Tarmunora) [reply code/obj/artifacts/artifactprocs.dm:268]: comedy option: let it go in but then detonate the bag a la putting a wand of cancellation in a bag of holding in nethack 😈

Comment (FlameArrow57) [reply code/obj/artifacts/artifactprocs.dm]: I went with tying it into the existing bag of holding negative effects, so adding a bag of holding or an activator key will do a bad effect

**Pull request merged**